test/v200/utils/common/project_test_utils.go

```diff
@@ -0,0 +1,244 @@
+	project := testDevfile.createProject(projectType)
+	testDevfile.SetProjectValues(&project)
+	return project.Name
+}
+
+// AddStarterProject adds a starter project of the specified type, with random attributes, to the devfile schema
+func (testDevfile *TestDevfile) AddStarterProject(projectType schema.ProjectSourceType) string {
+	starterProject := testDevfile.createStarterProject(projectType)
+	testDevfile.SetStarterProjectValues(&starterProject)
+	return starterProject.Name
+}
+
+// createProject creates a project of a specified type with only required attributes set
+func (testDevfile *TestDevfile) createProject(projectType schema.ProjectSourceType) schema.Project {
+	project := schema.Project{}
+	project.Name = GetRandomUniqueString(GetRandomNumber(8, 63), true)
+	LogInfoMessage(fmt.Sprintf("Create Project Name: %s", project.Name))
+
+	if projectType == schema.GitProjectSourceType {
+		project.Git = createGitProject()
+	} else if projectType == schema.GitHubProjectSourceType {
+		project.Github = createGithubProject()
+	} else if projectType == schema.ZipProjectSourceType {
+		project.Zip = createZipProject()
+	}
+	testDevfile.projectAdded(project)
+	return project
+}
+
+// generateStarterProject creates a starter project of a specified type with only required attributes set
+func (testDevfile *TestDevfile) createStarterProject(projectType schema.ProjectSourceType) schema.StarterProject {
+	starterProject := schema.StarterProject{}
+	starterProject.Name = GetRandomUniqueString(GetRandomNumber(8, 63), true)
+	LogInfoMessage(fmt.Sprintf("Create StarterProject Name: %s", starterProject.Name))
+
+	if projectType == schema.GitProjectSourceType {
+		starterProject.Git = createGitProject()
+	} else if projectType == schema.GitHubProjectSourceType {
+		starterProject.Github = createGithubProject()
+	} else if projectType == schema.ZipProjectSourceType {
+		starterProject.Zip = createZipProject()
+	}
+	testDevfile.starterProjectAdded(starterProject)
+	return starterProject
+
+}
+
+// createGitProject creates a git project structure with mandatory attributes set
+func createGitProject() *schema.GitProjectSource {
+	project := schema.GitProjectSource{}
+	project.Remotes = getRemotes()
+	return &project
+}
+
+// createGithubProject creates a github project structure with mandatory attributes set
+func createGithubProject() *schema.GithubProjectSource {
+	project := schema.GithubProjectSource{}
+	project.Remotes = getRemotes()
+	return &project
+}
+
+// createZipProject creates a zip project structure
+func createZipProject() *schema.ZipProjectSource {
+	project := schema.ZipProjectSource{}
+	return &project
+}
+
+// SetProjectValues sets project attributes, common to all projects, to random values.
+func (testDevfile *TestDevfile) SetProjectValues(project *schema.Project) {
+
+	if GetBinaryDecision() {
+		project.ClonePath = "./" + GetRandomString(GetRandomNumber(4, 12), false)
```

Comment:
I generally don't like using random values in tests since it could make the test runs non-reproducible. Could we just use a regular fake path here?

Reply:
I was requested to use a fuzzer to create values to be used in the tests and this was my way of doing it because real fuzzers include many characters which are invalid in the schema.  It is a trade off between testing as many different possible values to find weird errors and having totally reproducible tests. Partly to mitigate this I do write out all devfiles created to a tmp directory so you can see the values used in a failure, should a failure happen.

Reply:
Ah, okay. That makes sense then.